Sources/NIOCore/EventLoop.swift

```diff
@@ -436,49 +436,64 @@ public struct TimeAmount: Hashable, Sendable {
     /// - parameters:
     ///     - amount: the amount of microseconds this `TimeAmount` represents.
     /// - returns: the `TimeAmount` for the given amount.
+    ///
+    /// - note: returns `TimeAmount(.max)` if the amount, once converted to nanoseconds, exceeds `Int64.max`.
     @inlinable
     public static func microseconds(_ amount: Int64) -> TimeAmount {
-        return TimeAmount(amount * 1000)
+        let nanoseconds = amount &* 1000
+        return nanoseconds >= 0 ? TimeAmount(nanoseconds) : TimeAmount(.max)
```

Comment:
This isn't quite right: `TimeAmount` can represent a negative duration so `.microseconds(-1000)` would be incorrectly converted to `.max`. We'd also need to have unit tests for the extreme (overflow and underflow) cases to verify this behaves as expected.